Add self-managed cluster configuration (#971)
Azure/aad-pod-identity: website/content/en/docs/Getting started/role-assignment.md

  Your cluster will need the correct role assignment configuration to perform Azure-related operations.
---

Your cluster will need the correct role assignment configuration to perform Azure-related operations such as assigning and un-assigning the identity on the underlying VM/VMSS. You can run the following commands to help you set up the appropriate role assignments for your cluster identity before deploying aad-pod-identity.

AKS and aks-engine clusters require an identity to communicate with Azure. This identity can be either a **managed identity** (in the form of system-assigned identity or user-assigned identity) or a **service principal**. This section explains various role assignments that need to be performed before using AAD Pod Identity. Without the proper role assignments, your Azure cluster will not have the correct permission to assign and un-assign identities from the underlying virtual machines (VM) or virtual machine scale sets (VMSS). 

> In the case of self-managed clusters (manual installation of kubernetes on Azure VMs), you'll need to assign a **user managed identity** to the VM or VMSS or use a **service principal**.

Reviewer: ```suggestion
In the case of self-managed clusters (manual installation of Kubernetes on Azure VMs), you'll need to assign a **user-assigned managed identity** to the VM or VMSS or use a **service principal**. This is required for MIC to perform Azure-related operations for assigning/un-assigning the identity.
```

az role assignment create --role "Managed Identity Operator" --assignee <ID>  --scope /subscriptions/<SubscriptionID>/resourcegroups/<IdentityResourceGroup>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<IdentityName>
```

## User-assigned managed identities for self-managed clusters

> If you deploy yourself the VMs and install kubernetes, you'll need to assign the user-assigned managed identity to the underlying VMs.

Reviewer: ```suggestion
If you deploy the VMs and install Kubernetes instead of using tools like [aks-engine](https://github.com/Azure/aks-engine) or [capz](https://github.com/kubernetes-sigs/cluster-api-provider-azure), you'll need to assign the user-assigned managed identity to the underlying VMs.

```

### For VMSS

```bash 

> az vmss identity assign -n <VMSS name> -g <rg> --identities <IdentityName>

Reviewer: ```suggestion
az vmss identity assign -n <VMSS name> -g <rg> --identities <IdentityResourceID>

```

```

### For VMs

```bash

> az vm identity assign -n <VM name> -g <rg> --identities <IdentityName>

Reviewer: ```suggestion
az vm identity assign -n <VM name> -g <rg> --identities <IdentityResourceID>
```
